Adds x-directus-cache response header with HIT value

keesvanbemmel opened the pull request:
To be able to verify if data was retrieved from cache, besides response times.

This will add a response header named "x-directus-cache" with value "HIT" only if data was retreived from cache.

See https://github.com/directus/directus/discussions/12728

keesvanbemmel committed:
Adds x-directus-cache response header with HIT value

rijkvanzanten commented on api/src/middleware/cache.ts:
```diff
@@ -42,6 +42,7 @@ const checkCacheMiddleware: RequestHandler = asyncHandler(async (req, res, next)
 
 		res.setHeader('Cache-Control', getCacheControlHeader(req, cacheTTL));
 		res.setHeader('Vary', 'Origin, Cache-Control');
+		res.setHeader('x-directus-cache', 'HIT');
```
Lets only add the cache header when the cache is enabled through `CACHE_ENABLED`, and lets add another env var to control the name of the cache cache maybe? Not sure if that's too overkill, but I can imagine people might want to white label that header name 🤔 That flag would also allow for `false` to disable the header

keesvanbemmel replied:
Sure! Can add the label env flag. If not set, will not set the header.

As for the cache enabled: if it's not enabled that piece of code won't be reached right?

rijkvanzanten replied:
True! I was already thinking another step ahead in that we should also render a "MISS" when the cache is enabled but not used 🙂  In that case, it might also be hit when the cache is not used

keesvanbemmel committed:
Added ENV var for cache header and MISS, UNCACHEABLE

keesvanbemmel commented:
Something like this latest commit @rijkvanzanten ?

rijkvanzanten committed:
Merge branch 'main' into kees/add-cache-hit-response-header

rijkvanzanten commented:
Exactly like that last commit 💪🏻

rijkvanzanten committed:
Rename CACHE_HEADER_KEY -> CACHE_STATUS_HEADER

Tad opinionated, but I feel like it describes the functionality slightly better

rijkvanzanten committed:
Return miss consistently

Uncacheable doesn't quite describe what's happening. The user actively requests the cache to be skipped, which should result in an expected miss. The value isn't un-cacheable, the cache is skipped on purpose
